pkg/cmd/cli/describe/chaindescriber.go

```diff
@@ -0,0 +1,133 @@
+	"github.com/gonum/graph/topo"
+
+	osgraph "github.com/openshift/origin/pkg/api/graph"
+	buildedges "github.com/openshift/origin/pkg/build/graph"
+	buildgraph "github.com/openshift/origin/pkg/build/graph/nodes"
+	"github.com/openshift/origin/pkg/client"
+	imagegraph "github.com/openshift/origin/pkg/image/graph/nodes"
+	"github.com/openshift/origin/pkg/util/parallel"
+)
+
+// NotFoundErr is returned when the imageStreamTag (ist) of interest cannot
+// be found in the graph. This doesn't mean though that the ist does not
+// exist. A user may have an image stream without a build configuration
+// pointing at it. In that case, the ist of interest simply doesn't have
+// other dependant ists
+type NotFoundErr string
+
+func (e NotFoundErr) Error() string {
+	return fmt.Sprintf("couldn't find image stream tag: %q", string(e))
+}
+
+// ChainDescriber generates extended information about a chain of
+// dependencies of an image stream
+type ChainDescriber struct {
+	client.Interface
+	Output string
+}
+
+// NewChainDescriber returns a new ChainDescriber
+func NewChainDescriber(c client.Interface, out string) *ChainDescriber {
+	return &ChainDescriber{Interface: c, Output: out}
+}
+
+// MakeGraph will create the graph of all build configurations and the image streams
+// they point to via image change triggers in the provided namespace
+func (d *ChainDescriber) MakeGraph(namespace string) (osgraph.Graph, error) {
+	g := osgraph.New()
+
+	loaders := []GraphLoader{
+		&bcLoader{namespace: namespace, lister: d.Interface},
+	}
+	loadingFuncs := []func() error{}
+	for _, loader := range loaders {
+		loadingFuncs = append(loadingFuncs, loader.Load)
+	}
+
+	if errs := parallel.Run(loadingFuncs...); len(errs) > 0 {
+		return g, utilerrors.NewAggregate(errs)
+	}
+
+	for _, loader := range loaders {
+		loader.AddToGraph(g)
+	}
+
+	buildedges.AddAllInputOutputEdges(g)
+
+	return g, nil
+}
+
+// Describe returns the output of the graph starting from the provided
+// image stream tag (name:tag)
+func (d *ChainDescriber) Describe(namespace, name, tag string) (string, error) {
+	g, err := d.MakeGraph(namespace)
+	if err != nil {
+		return "", err
+	}
+
+	// Retrieve the imageStreamTag node of interest
+	ist := g.Find(imagegraph.ImageStreamTagNodeName(imagegraph.MakeImageStreamTagObjectMeta(namespace, name, tag)))
+	if ist == nil {
+		return "", NotFoundErr(fmt.Sprintf("%s:%s", name, tag))
+	}
+
+	// Filter out all but BuildConfig and ImageStreamTag nodes
+	nodeFn := osgraph.NodesOfKind(buildgraph.BuildConfigNodeKind, imagegraph.ImageStreamTagNodeKind)
+	// Filter out 1) all but BuildInputImage and BuildOutput edges and
+	// 2) inbound edges to the image stream tag of interest
+	allowedKinds := util.NewStringSet(buildedges.BuildInputImageEdgeKind, buildedges.BuildOutputEdgeKind)
+	var edgeFn osgraph.EdgeFunc = func(g osgraph.Interface, head, tail graph.Node, edgeKinds util.StringSet) bool {
+		return allowedKinds.HasAny(edgeKinds.List()...) && ist != tail
```

Comment:
> @deads2k removed the RemoveEdge method and added this instead (&& ist != tail). Is this what we want?

I think I like it better.  Can you promote this to a non-anonymous function?  `RemoveInboundEdges` that takes a `[]Node`?